### ISSUE TITLE
unknown error: window[requireName] is not a function

### ISSUE THREAD

**Issue by wojtkowiak** [electron/spectron#49]
Hey,
I am trying to check the title of my second window. Is this a bug, or am I doing something wrong?

```javascript
let handles = await app.client.windowHandles();
console.log(handles);
/*
{ sessionId: '0d66b5bac77357c36ea41fd298e55854',
  status: 0,
  value:
   [ 'CDwindow-97CD6BD5-C47A-45C1-9A66-7D0FF3B2F9DF',
     'CDwindow-946E9CE7-9A1C-458A-9DE2-00A975ACC5BD' ] }
*/
await app.client.windowByIndex(1);
console.log(await app.client.browserWindow.getTitle());
```
BTW `windowHandles` is missing in the readme. Also I saw on stackoverflow that somebody was doing `windowByIndex(handles.values[1])` but this also fails for me.

What is also weird is that my second window actually has focus, but `app.client.browserWindow.getTitle()` will get me the title of the first window. How can I make `app.client.browserWindow` to point at currently focused window?

**Comment by kevinsawicki** [comment]
> BTW windowHandles is missing in the readme.

The `app.client` API is currently documented at http://webdriver.io/api.html

**Comment by kevinsawicki** [comment]
> console.log(await app.client.browserWindow.getTitle());

This should be this instead:

```js
console.log(await app.browserWindow.getTitle());
```

**Comment by kevinsawicki** [comment]
> unknown error: window[requireName] is not a function

Do you have more of the stack trace that raised this error?

**Comment by kevinsawicki** [comment]
Also, does your app have `nodeIntegration` enabled in both windows?

**Comment by kevinsawicki** [comment]
Also, does `app.client.getTitle()` work for you?

**Comment by wojtkowiak** [comment]
Thank you very much for your time.
`unknown error: window[requireName] is not a function` unfortunately this did not appeared with a stack trace (I am using ava). 

Thanks to your hints I made it work. The problem was `nodeIntegration` being disabled indeed. So it seems that this error was actually triggered by `windowByIndex`.

I changed `app.client.browserWindow.getTitle()` to `app.browserWindow.getTitle()` and it works. `app.client.getTitle()` works also.

So after all maybe adding a note in readme about `nodeIntegration` would be helpful. 

I have one more question if I may - can I talk to the main process ipc directly from tests? I am trying to trigger some things directly from the tests instead of the renderer process.
I have tried `const ipc = app.electron.ipcRenderer;` but `ipc.send` does not seem to send to the main process. Is there any way to do that or do I just have to setup another ipc connection myself?

**Comment by kevinsawicki** [comment]
> So after all maybe adding a note in readme about nodeIntegration would be helpful.

Good point, I added https://github.com/electron/spectron#node-integration today which hopefully makes thing clearer.

> can I talk to the main process ipc directly from tests?

It should work, I just uploaded a simple spec to test it and it appears to pass currently, https://github.com/electron/spectron/commit/6d36884f0329f3a187ae105c40b091fba98d83aa

**Comment by wojtkowiak** [comment]
Hmm `app.electron.ipcRenderer.send('ipc-event', 'test')` does not work for me. Not sure what is the reason.

**Comment by wojtkowiak** [comment]
Oh wait, got it. Seems that when you use `ipcRenderer` also in html that is loaded in your app it takes precedence and using it directly in tests fails then.

**Comment by kevinsawicki** [comment]
> Seems that when you use ipcRenderer also in html that is loaded in your app it takes precedence and using it directly in tests fails then. So it seems that it will only work if you somehow disable ipc in the renderer for test purposes.

Hmm, that is surprising, could you include the code to help me reproduce this locally?

**Comment by wojtkowiak** [comment]
Good that you have asked. 
I've forked spectron and changed the `fixtures/app` just to find out that it works. Now I've reverted my changes from yesterday and it also works :smile: 
So it seems that I had some other problem somewhere which I probably fixed with other changes. It started working when I removed `var ipc = require('electron').ipcRenderer;` from my `index.html` but it seems that this was only coincidence.
Tried to nail it but my IDE did not keeps so detailed track of local changes.
So anyway, thank you for your time once again!
